### PR TITLE
picocom: 3.1 -> 3.2a

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Upstream (untagged) release.

Also:
- write history to $HOME/.cache instead of $HOME
- drop wrapper and patch in location of binaries
- add bash completion support
- start using shell helpers instead of manually copying in man page

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/picocom/default.nix [hunk 1]
```diff
@@ -1,36 +1,50 @@
-{ stdenv, fetchFromGitHub, makeWrapper, lrzsz, IOKit }:
+{ stdenv
+, fetchFromGitHub
+, installShellFiles
+, lrzsz
+, IOKit ? null
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/picocom/default.nix:5]:
```suggestion
, IOKit
```

File: pkgs/tools/misc/picocom/default.nix [hunk 2]
```diff
@@ -1,36 +1,50 @@
...
+}:
 
 assert stdenv.isDarwin -> IOKit != null;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/picocom/default.nix:8]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107379` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ino</li>
    <li>picocom</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107379` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>picocom</li>
  </ul>
</details>

File: pkgs/tools/misc/picocom/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/tools/misc/picocom/default.nix:5]: What's the reason for this change?

Comment (SuperSandro2000) [reply pkgs/tools/misc/picocom/default.nix:5]: There is no reason to do this.

**Pull request merged**